src/components/Page/components/Header/Header.tsx

```diff
@@ -138,159 +109,80 @@ class Header extends React.PureComponent<CombinedProps, State> {
+      ) : null;
 
-    if (secondaryActions.length === 0 && actionGroups.length === 0) {
-      return null;
-    }
+    const headerClassNames = classNames(
+      styles.Header,
+      titleHidden && styles.titleHidden,
+      separator && styles.separator,
+      navigationMarkup && styles.hasNavigation,
+      menuMarkup && styles.hasMenu,
+      mobileView && styles.mobileView,
+    );
 
-    const secondaryActionMarkup =
-      secondaryActions.length > 0
-        ? secondaryActionsFrom(secondaryActions)
-        : null;
+    return (
+      <div className={headerClassNames}>
+        {navigationMarkup}
 
-    const actionGroupsMarkup =
-      actionGroups.length > 0
-        ? actionGroups.map(({title, icon, actions, details}, index) => (
-            <div
-              className={styles.IndividualAction}
-              key={`ActionGroup-${title}-${index}`}
-            >
-              <ActionGroup
-                title={title}
-                icon={icon}
-                actions={actions}
-                details={details}
-                onOpen={this.handleActionGroupOpen}
-                onClose={this.handleActionGroupClose}
-                active={title === openActionGroup}
-              />
-            </div>
-          ))
-        : null;
+        <div className={styles.MainContent}>
+          <div className={styles.TitleMenuWrapper}>
+            {titleMarkup}
+            {menuMarkup}
+          </div>
 
-    return (
-      <div className={styles.SecondaryActions}>
-        <div className={styles.IndividualActions}>
-          {secondaryActionMarkup}
-          {actionGroupsMarkup}
+          {primaryActionMarkup}
         </div>
-      </div>
-    );
-  };
-
-  private handleRollupToggle = () => {
-    this.setState(({rollupOpen}) => ({rollupOpen: !rollupOpen}));
-  };
 
-  private handleActionGroupClose = (group: string) => {
-    this.setState(
-      ({openActionGroup}) =>
-        openActionGroup === group ? {openActionGroup: undefined} : {},
+        <EventListener event="resize" handler={this.handleResize} passive />
+      </div>
     );
-  };
+  }
 
-  private handleActionGroupOpen = (group: string) => {
-    this.setState({openActionGroup: group});
+  private handleToggleMobile = () => {
+    const {mobileView} = this.state;
+    this.setState({mobileView: !mobileView});
   };
 }
 
-function convertActionGroupToActionListSection({
-  title,
-  actions,
-}: ActionGroupDescriptor) {
-  return {title, items: actions};
-}
-
-function secondaryActionsFrom(
-  actions: SecondaryAction[],
-): ReadonlyArray<JSX.Element> {
-  return actions.map(({content, ...action}, index) => (
-    <div className={styles.IndividualAction} key={`Action-${content || index}`}>
-      <Action {...action}>{content}</Action>
-    </div>
-  ));
+// TODO: Can we instead get this from the <Frame />?
+// Or perhaps store in Context to be shared across components?
+function isMobileView(): boolean {
+  return navigationBarCollapsed().matches;
```

Comment:
Why do you feel it's necessary to move this out of this component and into context?

Reply:
Maybe its not - I was initially concerned about too many components repeating the exact same code. I'm fine to remove the comment and leave it.... less work 😅